src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,52 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int length = 0;
+        int[] diagonal = new int[matrix.length];
+
+        while (length < matrix.length) {
+            diagonal[length] = matrix[length][length];
+            length++;
+        }
+
+        return matrix == null ? null : diagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int height = matrix.length - 1;
+        int length = 0;
+        int[] diagonal = new int[matrix.length];
+
+        while (length < matrix.length && height >= 0) {
```

Comment:
it's better to create for loop here, so you won't create height and length variable outside of loop ( you don't use it anywhere outside of loop)